aten/src/ATen/native/cpu/Activation.cpp

```diff
@@ -632,6 +632,45 @@ void silu_backward_kernel(TensorIterator& iter) {
       });
 }
 
+
+void mish_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(
+      kBFloat16, iter.dtype(), "mish_cpu", [&]() {
+        const Vec256<scalar_t> kOneVec(scalar_t(1));
+        cpu_kernel_vec(
+            iter,
+            [](scalar_t x) {
+                return x * std::tanh(std::log1p(std::exp(x)));
+            },
+            [kOneVec](Vec256<scalar_t> x_vec) {
+              return x_vec * x_vec.exp().log1p().tanh();
+            });
+      });
+}
+
```

Comment:
Hi @Adnios, I am still understanding the PyTorch code architecture and came across your commit which is really interesting. Could you please explain me why do we have a mish_kernel and a mish_backward_kernel ? and why did you define math_mish_backward. Thank you.

Reply:
Hello, @venkateshtata. Sorry for the delay.
- `mish_kernel` and `mish_backward_kernel` are the C++ implementations of `mish` which are used by `torch._C._nn.mish` in `torch/nn/functional.py`
- `math_mish_backward` may be not necessary for is commit. But there may be a potential problem. See: https://github.com/pytorch/pytorch/pull/49439 https://github.com/pytorch/xla/issues/2652

I refer to a lot of codes of `softplus` and `silu` when I implement `mish`.
The blog of ezyang is very helpful to understand the PyTorch. http://blog.ezyang.com/2019/05/pytorch-internals/
Hope this helps.

Reply:
Thank you